src/libs/LibTokenizedVaultStaking.sol

```diff
@@ -82,6 +82,10 @@ library LibTokenizedVaultStaking {
     function _payReward(bytes32 _stakingRewardId, bytes32 _entityId, bytes32 _rewardTokenId, uint256 _rewardAmount) internal {
         AppStorage storage s = LibAppStorage.diamondStorage();
 
+        if (s.stakingConfigs[_entityId].initDate == 0) {
```

Comment:
Isn't this what we want? 🤔 
```suggestion
        if (s.stakingConfigs[_entityId].initDate == 0 || s.stakingConfigs[_entityId].initDate > block.timestamp) {
```

Reply:
> Isn't this what we want? 🤔

umm there is another check a few lines down for this which returns a more specific error. I think its better to separate these checks as we want the error messages to be clear on why the tx was reverted

Reply:
Good point, ignore my comment.